File: robot-return-to-origin.swift
// 执行用时：40 ms, 在所有 Swift 提交中击败了86.36%的用户
// 内存消耗：20.5 MB, 在所有 Swift 提交中击败了60.00%的用户
class Solution {
    func judgeCircle(_ moves: String) -> Bool {
        if moves.count & 1 == 1 {
            return false
        }
        var x = 0, y = 0
        for m in moves {
            switch m {
            case "L": x-=1
            case "R": x+=1
            case "U": y+=1
            case "D": y-=1
            default:
                break
            }
        }
        return x == 0 && y == 0
    }
}

File: robot-return-to-origin.kt

/**
 * 执行用时：292 ms, 在所有 Kotlin 提交中击败了16.67%的用户
 * 内存消耗：34.6 MB, 在所有 Kotlin 提交中击败了33.33%的用户
 */
class Solution {
    fun judgeCircle(moves: String): Boolean {
        return if (moves.length and 1 == 1) false else {
            var x = 0
            var y = 0
            moves.forEach { when (it) {
                'L' -> x--
                'R' -> x++
                'U' -> y++
                'D' -> y--
            } }
            return x == 0 && y == 0
        }
    }
}

